File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.io.*;
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for(int i=0; i<n; i++){
            String tmp1 = Integer.toBinaryString(arr1[i]);
            String tmp2 = Integer.toBinaryString(arr2[i]);
            
            if(tmp1.length() < n){
                tmp1 = String.format("%"+n+"s", tmp1).replace(' ', '0');
            }
            if(tmp2.length() < n){
                tmp2 = String.format("%"+n+"s", tmp2).replace(' ', '0');
            }
            String tmp = "";
            for(int j=0; j<n; j++){
                if(tmp1.charAt(j) == '1' || tmp2.charAt(j) == '1'){
                    tmp += "#";
                }else{
                    tmp += " ";
                }
            }
            answer[i] = tmp;
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    # 배열로 만들기
    for i in range(0, len(arr1)):
        list1 = list(format(arr1[i], f'0{n}b'))
        list2 = list(format(arr2[i], f'0{n}b'))
        # 1인지 0인지 판단하고 지도 만들기
        tmp = ''
        for j in range(0, len(list1)):
            if list1[j] == '1' or list2[j] == '1':
                list1[j] = 1
            tmp += str(list1[j])
            
        # 1은 #으로, 0은 공백으로 대치
        tmp = tmp.replace('1','#',n)
        tmp = tmp.replace('0',' ',n)
        answer.append(tmp)
    return answer